jwst/pipeline/calwebb_spec2.py

```diff
@@ -43,179 +48,226 @@ class Spec2Pipeline(Pipeline):
+            A Level2b association product.
+        """
+
+        # Find all the member types in the product
+        members_by_type = defaultdict(list)
+        for member in exp_product['members']:
+            members_by_type[member['exptype']].append(member['expname'])
+
+        # Get the science member. Technically there should only be
+        # one. We'll just get the first one found.
+        science = members_by_type['SCIENCE']
+        if len(science) != 1:
+            log.warn(
+                'Wrong number of science files found in {}'.format(
+                    exp_product['name']
+                )
+            )
+        science = science[0]
+
+        self.log.info('Working on input %s ...', science)
+        if isinstance(science, datamodels.DataModel):
+            input = science
+        else:
+            input = datamodels.open(science)
+        exp_type = input.meta.exposure.type
 
-            # Produce a resampled product, either via resample_spec for
-            # "regular" spectra or cube_build for IFU data. No resampled
-            # product is produced for time-series modes.
-            if input.meta.exposure.type in ['NRS_FIXEDSLIT', 'NRS_BRIGHTOBJ',
-                'NRS_MSASPEC', 'NIS_WFSS', 'NRC_GRISM']:
+        # Apply WCS info
+        input = self.assign_wcs(input)
 
-                # Call the resample_spec step
-                resamp = self.resample_spec(input)
+        # If assign_wcs was skipped, abort the rest of processing,
+        # because so many downstream steps depend on the WCS
+        if input.meta.cal_step.assign_wcs == 'SKIPPED':
+            log.error('Assign_wcs processing was skipped')
+            log.error('Aborting remaining processing for this exposure')
+            log.error('No output product will be created')
+            return
 
-                # Save the resampled product
-                if self.resample_spec.skip != True:
-                    self.save_model(resamp, 's2d')
-                    log.info('Saved resampled product to %s' % resamp.meta.filename)
+        # Do background processing, if necessary
+        if len(members_by_type['BACKGROUND']) > 0:
 
-                # Pass the resampled data to 1D extraction
-                x1d_input = resamp.copy()
-                resamp.close()
+            # Call the background subtraction step
+            input = self.bkg_subtract(input, members_by_type['BACKGROUND'])
 
-            elif exp_type in ['MIR_MRS', 'NRS_IFU']:
+            # Save the background-subtracted product, if requested
+            if self.save_bsub:
+                if isinstance(input, datamodels.CubeModel):
+                    self.save_model(input, "bsubints")
+                else:
+                    self.save_model(input, "bsub")
+
+        # Apply NIRSpec MSA imprint subtraction
+        # Technically there should be just one.
+        # We'll just get the first one found
+        imprint = members_by_type['IMPRINT']
```

Comment:
Yeah, see here we're using the label "IMPRINT", not "LEAKCAL". So the earlier instance of LEAKCAL in the asn properties should be changed to IMPRINT.